### PR TITLE
[stable/3.0] Invoke cluster health checks

### PR DESCRIPTION
**Pull request by jsuchome**
additionally added one commit from https://github.com/crowbar/crowbar-core/pull/619

### CODE REVIEW

**Comment by jsuchome** [comment]
The report should be probably shown differently than just into the log file. Not sure what would match the api conventions though...

**Comment by rsalevsky** [comment]
I think on C6 side this is fine enough.

**Comment by jsuchome** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;sKTTY1Wp&#x2F;65-3-s46p13-implement-api-for-first-step-of-upgrade-checks"> (3) S46P13: Implement API for first step of upgrade: checks</a>

File: crowbar_framework/app/models/api/crowbar.rb [hunk 1]
```diff
@@ -80,6 +80,15 @@ def maintenance_updates_installed?
       end
     end
 
+    def clusters_healthy?
+      service_object = CrowbarService.new(Rails.logger)
+      cluster_health = service_object.check_cluster_health
+      unless cluster_health.empty?
+        Rails.logger.warn("clusters report some problems: #{cluster_health}")
```
Comment (vuntz) [inline crowbar_framework/app/models/api/crowbar.rb]: I think `cluster_health` is not a string, so not sure if we should use it like this in the log.

File: crowbar_framework/app/models/crowbar_service.rb [hunk 2]
```diff
@@ -340,7 +340,7 @@ def check_cluster_health
         crm_failures[n.name] = ssh_retval[:stdout]
         next
       end
-      ssh_retval = n.run_ssh_cmd('crm status | grep "^Failed actions:"')
+      ssh_retval = n.run_ssh_cmd("crm status | grep -A 2 '^Failed Actions:'")
```
Comment (vuntz) [inline crowbar_framework/app/models/crowbar_service.rb]: Don't we need some `LANG=C` or something like this, just to be safe?

Comment (jsuchome) [inline crowbar_framework/app/models/crowbar_service.rb]: Right. I was assuming it is already part of run_ssh_cmd, but it is not

File: crowbar_framework/app/models/api/crowbar.rb (hunk 1 above)
Comment (jsuchome) [inline crowbar_framework/app/models/api/crowbar.rb]: It's a hash describing various types of errors. Is anything wrong with logging that?

Comment (vuntz) [inline crowbar_framework/app/models/api/crowbar.rb]: Well, I prefer nice strings in the logs, instead of data structures :-)

Comment (jsuchome) [inline crowbar_framework/app/models/api/crowbar.rb]: See how the hash could look like. 
https://github.com/crowbar/crowbar-core/blob/stable/3.0/crowbar_framework/app/models/crowbar_service.rb#L319
The original idea probably was that the api somehow passes this to UI which can show it the nice way. I'd say logging such hash is good enough, but ok...

**Comment by jsuchome** [comment]
This is actually not invoked anywhere, not sure if I should wait for green gating...

**Comment by nicolasbock** [comment]
+1

**Comment by MaximilianMeister** [comment]
+1

**Pull request merged**